### PR TITLE
Adding function to get user by username

### PR DESCRIPTION
**Pull request by EricTendian**

### CODE REVIEW

**Comment by EricTendian** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;h7FI5IZq&#x2F;9-0-5-answers-api-v1-community-list-endpoint-giving-a-500-on-production"> (0.5) Answers API v1 community list endpoint giving a 500 on production</a>

File: src/AnswerbaseClient.php [hunk 1]
```diff
@@ -203,6 +203,14 @@ public function getUserByEmail($email = '')
     /**
      * @codeCoverageIgnore
      */
+    public function getUserByUsername($email = '')
```
Comment (karllhughes) [inline src/AnswerbaseClient.php]: Shouldn't we call this input `$username` not `$email`? Seems like a clearer variable name. @EricTendian

**Comment by karllhughes** [comment]
Also @EricTendian please add a test for this new functionality.

Thanks!

**Comment by EricTendian** [comment]
@karllhughes copy-paste fail. Fixed that and added some more tests.

**Pull request merged**